rust/rsc/src/rsc/main.rs

```diff
@@ -616,6 +618,56 @@ mod tests {
                 "obytes":1000
             })
         );
+
+        // Protected post blob route without auth should 401
+        let body_data: Vec<u8> = Vec::new();
+        let res = router
+            .call(
+                Request::builder()
+                    .uri("/blob")
+                    .method(http::Method::POST)
+                    .header(
+                        CONTENT_TYPE,
+                        format!("multipart/form-data; boundary={}", BOUNDARY),
+                    )
+                    .body(body_data.into())
+                    .unwrap(),
+            )
+            .await
+            .unwrap();
+
+        assert_eq!(res.status(), StatusCode::UNAUTHORIZED);
+
+        // Protected post blob route with auth should 200
+        let mut body_data: Vec<u8> = Vec::new();
+        write!(body_data, "--{}\r\n", BOUNDARY).unwrap();
+        write!(
+            body_data,
+            "Content-Disposition: form-data; name=\"file\";\r\n"
+        )
+        .unwrap();
+        write!(body_data, "\r\n").unwrap();
+        write!(body_data, "contents").unwrap();
+        write!(body_data, "\r\n").unwrap();
+        write!(body_data, "--{}--\r\n", BOUNDARY).unwrap();
+
+        let res = router
+            .call(
+                Request::builder()
+                    .uri("/blob")
+                    .method(http::Method::POST)
+                    .header(
+                        CONTENT_TYPE,
+                        format!("multipart/form-data; boundary={}", BOUNDARY),
+                    )
+                    .header(AUTHORIZATION, api_key)
```

Comment:
Are you missing an `clone` on the `api_key`? Or do we not need one because this is the last request?

Reply:
I don't need one here because its the last thing to take ownership of `api_key`. Rust wouldn't compile if a clone was needed here :)